forms-flow-web/src/components/Form/Item/FlowEdit.js

```diff
@@ -54,40 +55,63 @@ const FlowEdit = forwardRef((_, ref) => {
 
 
   const saveFlow = async () => {
-    const bpmnModeler = bpmnRef.current?.getBpmnModeler();
-    const xml = await createXMLFromModeler(bpmnModeler);
-
-    //if xml is same as existing process data, no need to update
-    const isEqual = await compareXML(processData?.processData, xml);
-    if (isEqual) {
-      toast.success(t("Process updated successfully"));
-      return;
-    }
-    if (!validateProcess(xml)) {
-      return;
-    }
-    
-    updateProcess({ type:"BPMN", id: processData.id, data:xml }).then((response) => {
+    try{
+      const bpmnModeler = bpmnRef.current?.getBpmnModeler();
+      const xml = await createXMLFromModeler(bpmnModeler);
+  
+      //if xml is same as existing process data, no need to update
+      const isEqual = await compareXML(processData?.processData, xml);
+      if (isEqual) {
+        toast.success(t("Process updated successfully"));
+        return;
+      }
+      if (!validateProcess(xml)) {
+        return;
+      }
+      setSavingFlow(true);
+      const response = await updateProcess({ type:"BPMN", id: processData.id, data:xml });
       dispatch(setProcessData(response.data));
       toast.success(t("Process updated successfully"));
-    });
+      setSavingFlow(false);
+    }catch(error){
+      setSavingFlow(false);
+      toast.error(t("Failed to update process"));
+    }
   };
 
   //handle discard changes
-  const handleDiscard = ()=>{
+  const handleDiscardConfirm = ()=>{
     if(bpmnRef.current){
       //import the existing process data to bpmn
       bpmnRef.current?.handleImport(processData?.processData);
+      handleHanldeDisacardModal();
     }
 };
 
+ 
+
   useImperativeHandle(ref, () => ({
     saveFlow,
   }));
 
+ 
+
   return (
     <Card>
-
+      <ConfirmModal
+        show={showDiscardModal}
+        title={t(`Are you Sure you want to Discard Flow Changes`) }
+        message={ <>
+        <p>{t("Are you sure you want to discard all the changes of the Flow?")}</p>
+        <p className='fw-bold'>{t("This action cannot be undone.")}</p>
+        </> }
```

Comment:
Can we pass this as 2 props . We are having 'message' and 'secondaryMessage' available in confirmModal